File: backend/src/main/java/de/interaapps/pastefy/model/requests/paste/EditPasteRequest.java
package de.interaapps.pastefy.model.requests.paste;

import de.interaapps.pastefy.model.database.Paste;

import java.util.List;

public class EditPasteRequest {

    public String title;
    public String content;
    public Boolean encrypted;
    public String folder;
    public Paste.Type type;

    public List<String> tags;

    public Paste.Visibility visibility;

    public String expireAt;
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/dto/pastes/EditPasteRequest.kt
package de.interaapps.pastefy.dto.pastes

import de.interaapps.pastefy.enums.PasteType
import de.interaapps.pastefy.enums.PasteVisibility

data class EditPasteRequest(
    val title: String? = null,
    val content: String? = null,
    val encrypted: Boolean? = null,
    val folder: String? = null,
    val expireAt: String? = null,
    val tags: List<String>? = null,
    val visibility: PasteVisibility? = null,
    val type: PasteType? = null,
)
